app/src/main/java/com/connfa/injection/ActivityLifecycle.java

```diff
@@ -0,0 +1,14 @@
+package com.connfa.injection;
+
+import javax.inject.Scope;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+@Scope
+@Documented
+@Retention(SOURCE)
+public @interface ActivityLifecycle {
```

Comment:
I hate the term `Singleton` because it's got its own meaning, so usually just create a scope for the activities and one for the application :)